src/Hint/Context.hs

```diff
@@ -157,7 +181,7 @@ removePhantomModule pm =
        if safeToRemove
          then do mayFail $ do res <- runGhc1 GHC.load GHC.LoadAllTargets
                               return $ guard (isSucceeded res) >> Just ()
-                 liftIO $ removeFile (pmFile pm)
+              `finally` do liftIO $ removeFile (pmFile pm)
```

Comment:
Btw, I kind of snuck this in... this fixes one case of hint leaving temporary files behind. I don't see a bug report for this.

Reply:
Oh, that's a nice catch. Please do this change as a separate commit, briefly explaining the bug. No need to add a test nor open an issue, since it's a minor thing and a small fix.